FirebaseDataConnect/Tests/Integration/IntegrationTests.swift

```diff
@@ -12,10 +12,218 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import Foundation
+import XCTest
 
+import FirebaseCore
 @testable import FirebaseDataConnect
 
-import XCTest
+final class IntegrationTests: XCTestCase {
+  class func setupFirebaseApp() {
+    if FirebaseApp.app() == nil {
+      let options = FirebaseOptions(googleAppID: "0:0000000000000:ios:0000000000000000",
+                                    gcmSenderID: "00000000000000000-00000000000-000000000")
+      options.projectID = "fdc-test"
+      FirebaseApp.configure(options: options)
+    }
+  }
+
+  private let setupQ = DispatchQueue(
+    label: "setupFDCIntegrationsTests",
+    autoreleaseFrequency: .workItem
+  )
+  private var _setupComplete = false
+  private var setupComplete: Bool {
```

Comment:
Could you explain the usage of DispatchQueue here, instead of just setting/getting a boolean variable? Is this to ensure thread safety?

Reply:
Yes, the idea was to only call the configure once. Yes, its for thread safety. The setup gets called before every testXyz() function. 

I am going to consider putting this in a static in a later change. But I haven't found a good pattern.

Reply:
Why not do all the `once` things in `class func setUp` ?

Reply:
The one time setup is an http call to the emulator so its asynchronous. The class func doesn't have a signature that works async. 

The setup(completion:) can be used to indicate via the callback when setup is complete. 

I also tried the setup() async throws (not the class func) but for some reason it isn't getting called by XCTest. So I ended up using the completion block equivalent.

Reply:
I was able to use a singleton actor so I got rid of all the dispatch code.